cmd/validators/main.go

```diff
@@ -0,0 +1,287 @@
+	var (
+		tmAddrFl = flag.String("tm", "https://bov.NETWORK.iov.one:443", "Tendermint node address. Use proper NETWORK name.")
+		pkPathFl = flag.String("pubkey", "pubkey.json", "Path to JSON encoded, public key of the validator.")
+		hexKeyFl = flag.String("key", "0a40d34c1970ae90acf3405f2d99dcaca16d0c7db379f4beafcfdf667b9d69ce350d27f5fb440509dfa79ec883a0510bc9a9614c3d44188881f0c5e402898b4bf3c9", "Hex encoded, private key used to sign the transaction.")
+		powerFl  = flag.Int64("power", 10, "Validator node power")
+	)
+	flag.Parse()
+
+	publicKey, pkType, err := readPublicKey(*pkPathFl)
+	if err != nil {
+		return fmt.Errorf("cannot read public key: %s", err)
+	}
+
+	genesis, err := fetchGenesis(*tmAddrFl)
+	if err != nil {
+		return fmt.Errorf("cannot fetch genesis: %s", err)
+	}
+
+	// This tool is meant to be agnostic of network implementation. We need
+	// a transaction object though. Both bns and bcp have binary compatible
+	// transaction declaration, so we can use any of them for both, as long
+	// as they support used extension.
+	tx := app.Tx{
+		Sum: &app.Tx_SetValidatorsMsg{
+			SetValidatorsMsg: &validators.SetValidatorsMsg{
+				ValidatorUpdates: []*validators.ValidatorUpdate{
+					{
+						Pubkey: validators.Pubkey{
+							Type: pkType,
+							Data: publicKey,
+						},
+						Power: *powerFl,
+					},
+				},
+			},
+		},
+	}
+
+	rawPrivateKey, err := hex.DecodeString(*hexKeyFl)
+	if err != nil {
+		return fmt.Errorf("cannot hex decode private key: %s", err)
+	}
+	var key crypto.PrivateKey
+	if err := key.Unmarshal(rawPrivateKey); err != nil {
+		return fmt.Errorf("cannot deserialize private key: %s", err)
+	}
+
+	cli := rpcclient.NewHTTP(*tmAddrFl, "/websocket")
+
+	seq, err := addrSeq(cli, key.PublicKey().Address())
+	if err != nil {
+		return fmt.Errorf("cannot fetch address sequence: %s", err)
+	}
+	sig, err := sigs.SignTx(&key, &tx, genesis.ChainID, seq)
+	if err != nil {
+		return fmt.Errorf("cannot sign transaction: %s", err)
+	}
+	tx.Signatures = append(tx.Signatures, sig)
+
+	if err := broadcast(cli, &tx); err != nil {
+		const unauthorizedCode = 3
+		if e, ok := err.(*httpError); ok && e.Code == unauthorizedCode {
+			log.Printf(
+				"Your signature address is %s\nMake sure to update 'update_validators' in genesis file.",
+				key.PublicKey().Address())
+		}
+		return fmt.Errorf("cannot broadcast: %s", err)
+	}
+	return nil
+}
+
+// addrSeq returns the next sequence number for given address.
+func addrSeq(cli rpcclient.Client, addr weave.Address) (int64, error) {
+	// TODO
+	return 3, nil
+	/*
+		resp, err := cli.ABCIQuery("/???", addr)
+		if err != nil {
+			return 0, fmt.Errorf("cannot query: %s", err)
+		}
+	*/
+}
+
+// readPublicKey returns public key encoded as JSON object.
+// Example content format:
+//   {"type": "ed25519",
+//    "value": "ASrHklGzkWYreMkjmhK9bwqUbDk1+1KflU+wpDAkvZs="}
+func readPublicKey(path string) (weave.Address, string, error) {
+	raw, err := ioutil.ReadFile(path)
```

Comment:
pubKey could be passed via stdin of env as described in the original spec. Then we don't have to deal with encryption/ decryption.